akka-actor/src/main/scala/akka/actor/ActorCell.scala

```diff
@@ -694,15 +694,18 @@ private[akka] class ActorCell(
       throw new IllegalArgumentException("ActorCell has no props field")
   }
 
-  final protected def clearActorFields(actorInstance: Actor, recreate: Boolean): Unit = {
-    setActorFields(actorInstance, context = null, self = if (recreate) self else system.deadLetters)
+  final protected def clearActorFields(actorInstance: Actor, recreate: Boolean, context: Option[ActorContext] = Some(null)): Unit = {
+    setActorFields(actorInstance, context = context, self = if (recreate) self else system.deadLetters)
```

Comment:
but there is a reason for why the context was nulled, to avoid accidental memory leaks.

I thought the conclusion in the ticket was that we should move away from the `import context.dispatcher` and use the more explicit `private implicit val ec: ExecutionContext = context.dispatcher`

Reply:
ok, I'll create another PR to fix the docs

Reply:
there might also be usages of import internally in Akka and those should also be changed

Reply:
Ouch. I'd point out that the import variety has been all over the docs for a while and as hinted at in the Google thread about it, it is the "done thing". That's likely a lot of existing code which uses `import context.dispatcher` (I know all mine does) so anything which can be done to make this safer rather than simply changing the docs would be worthwhile, I think.

Reply:
On the other hand I can't think of any code I've written which runs the risk of still using that execution context after the actor has died except where that would already have been an error state, so perhaps I'm exaggerating the magnitude.